core/transport/src/main/proto/grpc/common.proto

```diff
@@ -76,6 +76,7 @@ enum CommandType {
   Free = 3;     // Ask the worker to free files.
   Delete = 4;   // Ask the worker to delete files.
   Persist = 5;  // Ask the worker to persist a file for lineage
+  PreRegisterAndRegister = 6;  // Ask the worker to PreRegister
```

Comment:
If the "PreRegister" is embedded in the register logic, you don't need this extra command.

Reply:
When the ClusterId in the heartbeat is different from the current ClusterId: `PreRegisterAndRegister`

Reply:
Since the "pre-register" logic happens in GetWorkerId, I think you should call this command GetWorkerId. I don't want to define the "pre-register" concept here because we may want to use this word for other things.